src/hooks/useUpdateRecentSeatedUsers.ts

```diff
@@ -18,34 +19,63 @@ import { getCurrentTimeInMilliseconds } from "utils/time";
 
 import { useUser } from "hooks/useUser";
 
-export const useUpdateRecentSeatedUsers = <T extends VenueTemplate>(
+const updateSeatedData = async <T extends VenueTemplate>(
+  template: T,
+  venueId: string | undefined,
+  userId: string | undefined,
+  venueSpecificData: RecentSeatedUserData<T>["venueSpecificData"] | FalseyValue
+) => {
+  if (!venueSpecificData || !venueId || !userId) return;
+  console.log("updating");
+
+  const withTimestamp: RecentSeatedUserTimestamp<T> = {
+    template,
+    venueId,
+    venueSpecificData,
+    lastSittingTimeMs: getCurrentTimeInMilliseconds(),
+  };
+
+  return firebase
+    .firestore()
+    .collection("venues")
+    .doc(venueId)
+    .collection("recentSeatedUsers")
+    .doc(userId)
+    .set(withTimestamp);
+};
+
+const useUpdateRecentSeatedUsers = <T extends VenueTemplate>(
   template: T,
   venueId: string | undefined,
   venueSpecificData: RecentSeatedUserData<T>["venueSpecificData"] | FalseyValue
 ) => {
   const { userId } = useUser();
 
-  useInterval(() => {
-    if (!venueSpecificData || !venueId) return;
-
-    const withTimestamp: RecentSeatedUserTimestamp<T> = {
-      template,
-      venueId,
-      venueSpecificData,
-      lastSittingTimeMs: getCurrentTimeInMilliseconds(),
-    };
-
-    void firebase
-      .firestore()
-      .collection("venues")
-      .doc(venueId)
-      .collection("recentSeatedUsers")
-      .doc(userId)
-      .set(withTimestamp);
-  }, VENUE_RECENT_SEATED_USERS_UPDATE_INTERVAL);
+  const intervalRunning = Boolean(venueSpecificData && venueId);
+  useInterval(
+    () => {
+      void updateSeatedData(template, venueId, userId, venueSpecificData);
+    },
+    intervalRunning ? VENUE_RECENT_SEATED_USERS_UPDATE_INTERVAL : null
+  );
+
+  useEffect(() => {
+    void updateSeatedData(template, venueId, userId, venueSpecificData);
+  }, [template, userId, venueId, venueSpecificData]);
+};
```

Comment:
Given that it's useEffect, you have the option to return a function from here which should clean your previous position

Reply:
Good point!
We need to remove two pieces of data when user leaves their seat: `recentSeatedUsers` and the venue-specific data (`seatedSectionUsers` for auditorium and `seatedTableUsers` for jazzbar etc)
There are already hooks for removing venue-specific data in each of the venues (e.g `leaveSeat` in cleanup) but for additional safety I updated triggered functions to to remove venue-specific data when the user leaves seat too.
Commit https://github.com/sparkletown/sparkle/pull/2361/commits/82820f56e03473510aa8e855499a51e95e463eaa